# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/spellchecker/FilteredContentIterator.h
# Commit: 3601308f553b
# Full Hash: 3601308f553b3d381a68f6e90a971bd7cb7b1684
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-08-10 04:42:07
# Regressor Bug: 1845215
# File Overlap Count: 1
# Description:
#   Bug 1845215 - part 1: Make `ContentIteratorBase` a template class to work without strong pointers r=smaug
#   
#   `ContentEventHandler` works with `PostContentIterator` and `PreContentIterator`
#   when it scans DOM nodes in a range.  While iterating the DOM nodes, script
#   never runs.  Therefore, we can make `ContentEventHandler` work with new
# ==============================================================================

diff -r f2506aaa10bd -r 3601308f553b editor/spellchecker/FilteredContentIterator.h
--- a/editor/spellchecker/FilteredContentIterator.h	Wed Aug 09 23:05:26 2023 +0000
+++ b/editor/spellchecker/FilteredContentIterator.h	Thu Aug 10 00:59:28 2023 +0000
@@ -66,7 +66,7 @@
   void CheckAdvNode(nsINode* aNode, bool& aDidSkip, eDirectionType aDir);
   nsresult SwitchDirections(bool aChangeToForward);
 
-  ContentIteratorBase* MOZ_NON_OWNING_REF mCurrentIterator;
+  SafeContentIteratorBase* MOZ_NON_OWNING_REF mCurrentIterator;
   PostContentIterator mPostIterator;
   PreContentIterator mPreIterator;
 
